src/DetailPage.js

```diff
@@ -0,0 +1,79 @@
+import { useState, useEffect } from 'react';
+import { getRestaurantById, updateRestaurant } from './services/fetch-utils';
+import { useHistory, useRouteMatch, useParams } from 'react-router-dom';
+
+export default function DetailPage() {
+  const [restaurant, setRestaurant] = useState({});
+  const match = useRouteMatch();
+  const { push } = useHistory();
+  const [Name, setName] = useState('');
```

Comment:
generally speaking I would not recommend using capital letters for state items (camel case instead) but no need to fix and resubmit. Just a note for going forward.